wknml/nml_utils.py

```diff
@@ -1,45 +1,35 @@
 from typing import Union, Dict, List, Tuple
-from math import ceil
 from copy import deepcopy
-import networkx as nx
+from math import acos, ceil
 import numpy as np
+import networkx as nx
 
 from . import NML
 from .nml_generation import generate_graph, generate_nml
 
 
-def ensure_max_edge_length(nml_or_graph: Union[NML, Tuple[Dict[str, List[nx.Graph]], Dict]], max_length: int) -> Union[NML, nx.Graph]:
-    # it is easier to operate on a graph
-    if isinstance(nml_or_graph, NML):
-        nml_graph, parameter_dict = generate_graph(nml_or_graph)
-    else:
-        nml_graph = nml_or_graph[0]
-        parameter_dict = nml_or_graph[1]
-    max_id = detect_max_node_id_from_all_graphs(nml_graph)
-    next_valid_id = max_id + 1
-    for group in nml_graph.values():
-        for graph in group:
-            next_valid_id = ensure_max_edge_length_in_tree(graph, max_length, next_valid_id)
+def get_vector(node: Dict) -> np.ndarray:
+    return np.array(node["position"])
```

Comment:
I'm wondering whether this should be multiplied by the scale, since all the calculations should happen in nm space.

See https://github.com/scalableminds/wknml/issues/22
> Note that all that should happen in nm space.

also @georgwiese

Reply:
I don't know if the position does already include the scale since `position` is a list of 3 floats. What is the specification?

Reply:
I'd assume that the position contains voxel coordiantes (without scale), but you can simply test it, by creating a NML in wk (on some dataset where the scale is not 1,1,1) and downloading it. The positions you see in WK are voxel coordinates. If the same coordinates appear in your NML, the hypothesis is confirmed. In that case, you need to multiply the scale.

By the way, in general, it's a good idea to test your script with this workflow (create an NML with wk, add some nodes which match your distance criteria and then download, process the NML and re-upload and verify).